### ISSUE TITLE
[SUGGESTION] Condition Box Small block for compact logic and alternative functionality to logic gates

### ISSUE THREAD

**Issue by nokturnihs** [tsunamayo/Starship-EVO#3055]
This suggestion would require a LABEL field in most your existing logic components (timer, query, etc) - 

The idea - using a block similar in size to a ship computer, which accepts logic inputs and outputs, the player can use a single block to execute logic commands using simplistic IF/AND/OR/ELSE/THEN statements. This would require labels being applied to both inputs and outputs (these could be generated automatically) - then the player can access the condition box and use a simplistic syntax to execute logic. 

I realize this might be a bit of typing and the example below may not be perfect but frankly speaking the logic gate thing is REALLY hard for me personally to wrap my brain around while coding in lua, etc is pretty straight forward. Gets really tiring spending HOURS trying to figure out how to do some pretty basic stuff. This is probably not for everyone too but it doesnt break the existing logic system at all, rather it extends it. Besides that, it shrinks down logic for those individuals with small ships where extensive logic would come at a space premium. an example is below.

------------------------------------------------------------
For example: I want to make a Slide Rail move, then a piston and if both are fully extended rotate a hinge. If the toggle switch is pressed again then move the hinge first, then retract the piston, then the hinge.

The toggle would have a label "main", the query gate showing the state of the Slide rail is "railout" and the piston query gate is "pistonout". The hinge query gate is "hingeturned". The Timers for each are "slidemove", "pistonmove", and "hingeturner"

The following example is of a system that causes "armor plates" to slide off the roof of a vessel, then turn down to cover angled windows. Due to collision, if you just execute them all at the same time it doesn't work (or look right for that matter) - So in order for these "armor curtains" to work, when toggled on, the slide and piston needs to move then turn the hinge to cover the window. then when disabled, turn the hinge first and then retract the piston and rail.

The condition box might be something like this:

IF
main = TRUE
AND hingeturned = TRUE      ##The hinge is already turned
ACTIVATE hingeturner   ## Return the hinge to it's original position (off)
ENDIF

IF
main = TRUE
AND hingeturned = FALSE  ## The hinge is in it's original position so move the rail and piston
ACTIVATE slidemove
ACTIVATE pistonmove
ENDIF

IF
main = TRUE
AND hingturned = FALSE   ## The Hinge is in it's original position and the rail and piston has been extended so turn the hinge.
AND raliout = TRUE
AND pistonout = TRUE
ACTIVATE hingeturner
ENDIF

IF
main = FALSE
AND hingeturned = TRUE   ## The system has been turned off, hinge is turned and rail/piston extended, so move the hinge first
AND railout = TRUE
AND pistonout = TRUE
ACTIVATE hingeturner
ENDIF

IF
main = FALSE
AND hingeturned = FALSE  ## System turned off, Hinge has been turned so now retract the piston and sliderail.
AND railout = TRUE
AND pistonout = TRUE
ACTIVATE slidemove
ACTIVATE pistonmove
ENDIF
------------------------------------------------------------

Side Note: I have been messing with the logic for hours on this particular puzzle so if any logic gurus want to throw me a bone, won't complain. :)

**Comment by CurioInventorium** [comment]
I would be cool to have an ingame coding system, it'd be similar to what the people working on Starbase are doing.

I did a video on YouTube on sequenced activation of things, it was called "Four landing gears and a Hanger". I don't know if I'm allowed to post links for my channel here.
There have also been other videos, I think someone called it ABBA logic in the title. But I'm terrible at remembering names.

Also remember that logic blocks can be stacked on top of each other.

**Comment by MatCat** [comment]
Other people have suggested programming block and it was said it is planned in the future :).  That being said, if you can program getting into logic isn't THAT HARD, think of it is basic pseudo coding rather then logic,, for example if MAIN = true AND hinge = false, is really a simple AND gate with an input from main, and a nor gate (acting as a not gate) to invert the hinge output to a true into the AND gate, now you have a IF main AND !hinge expression, the output of the AND gate is your ACTIVATE, so link it to those things, you can use query blocks to know when a hinge or rotator or slider is to a certain position, though honestly I tend to just use timers for this, its rare that you have variable speed in these things so timers are easier often then query blocks for basic things like armor sliding around or landing gear.

**Comment by nokturnihs** [comment]
Yeah I dunno why I can't wrap my brain around it - I program (when I got a gig anyway, and it's not server recovery stuff) for a living - it just for some reason boggles and frustrates me a lot. Glad to hear it's coming someday, because I'd much prefer it over this kind of stuff. The same issue is why I just play modded minecraft and stuff. Learning disability or just getting old, dunno. Thanks for the tips though will try messing with it more.

**Comment by tsunamayo** [comment]
Yeah @nokturnihs I am with you, coding is just more natural than using gate... Thats a pretty big dev though, but was always something I wanted to do at one point. I have other higher priority item on the logic to-do list though (input override and event gate), so you will have to be patient ;)
Cheers
